api/pkg/apis/v1alpha1/model/summary.go

```diff
@@ -52,22 +52,21 @@ func (s *SummarySpec) UpdateTargetResult(target string, spec TargetResultSpec) {
 	if v, ok := s.TargetResults[target]; !ok {
 		s.TargetResults[target] = spec
 	} else {
-		status := v.Status
+		status := "OK"
+		maps.Copy(v.ComponentResults, spec.ComponentResults)
 		if spec.Status != "OK" {
 			status = spec.Status
-		}
-		message := v.Message
-		if spec.Message != "" {
-			if message != "" {
-				message += "; "
+		} else {
+			for _, componentStatus := range v.ComponentResults {
+				if componentStatus.Status != v1alpha2.Accepted {
+					status = v.Status
```

Comment:
We shouldn't override the status according to component result. The target status is determined by caller.